items/sign_host_keys.py

```diff
@@ -0,0 +1,145 @@
+    from sshkey_tools.keys import PrivateKey, PublicKey
+except ImportError:
+    raise bundlewrap.exceptions.BundleError("Please install package sshkey-tools>=0.9 first.")
+
+
+# See https://stackoverflow.com/a/49782093
+def remove_dir_recursive(path):
+    directory = Path(path)
+    for item in directory.iterdir():
+        if item.is_dir():
+            os.rmdir(item)
+        else:
+            item.unlink()
+    directory.rmdir()
+
+
+class SignHostKeys(Item):
+    """
+    Sign SSH Host Keys
+    """
+    BUNDLE_ATTRIBUTE_NAME = "sign_host_keys"
+    NEEDS_STATIC = [
+        "pkg_apt:",
+        "pkg_pacman:",
+        "pkg_yum:",
+        "pkg_zypper:",
+    ]
+    ITEM_ATTRIBUTES = {
+        'path': None,
+        'ca_password': None,
+        'ca_path': None,
+        'days_valid': 3650,
+    }
+    ITEM_TYPE_NAME = "sign_host_key"
+    REQUIRED_ATTRIBUTES = [
+        'path',
+        'ca_password',
+        'ca_path',
+    ]
+
+    def get_cert_path(self):
+        return self.attributes.get('path') + '.pub.crt'
+
+    @classmethod
+    def block_concurrent(cls, node_os, node_os_version):
+        """
+        Return a list of item types that cannot be applied in parallel
+        with this item type.
+        """
+        return []
+
+    def __repr__(self):
+        return "<Sign Host Key path:{} ca_path:{}>".format(self.attributes['path'],
+                                                                 self.attributes['ca_path'])
+
+    def cdict(self):
+        return {
+            f'{self.get_cert_path()} exist': True
+        }
+
+    def sdict(self):
+        path_info = PathInfo(self.node, self.get_cert_path())
+        return {
+            f'{self.get_cert_path()} exist': path_info.exists
+        }
+
+    def display_on_create(self, cdict):
+        """
+        Given a cdict as implemented above, modify it to better suit
+        interactive presentation when an item is created. If there are
+        any when_creating attributes, they will be added to the cdict
+        before it is passed to this method.
+
+        Implementing this method is optional.
+        """
+        return cdict
+
+    def display_dicts(self, cdict, sdict, keys):
+        """
+        Given cdict and sdict as implemented above, modify them to
+        better suit interactive presentation. The keys parameter is a
+        list of keys whose values differ between cdict and sdict.
+
+        Implementing this method is optional.
+        """
+        return (cdict, sdict, keys)
+
+    def display_on_delete(self, sdict):
+        """
+        Given an sdict as implemented above, modify it to better suit
+        interactive presentation when an item is deleted.
+
+        Implementing this method is optional.
+        """
+        return sdict
+
+    def fix(self, status):
+        tmpdir = mkdtemp()
+
+        pub_file_local = os.path.join(tmpdir, f'ssh_host_{self.attributes.get("key_format")}_key.pub')
```

Comment:
this will clash if it is run in paralell with other host. since this will generate ssh_host_rsa_key.pub for every host.

Reply:
Nope. `mkdtemp` is creating unique temporary dirs.